internal/controllers/clusterextension_controller_test.go

```diff
@@ -229,22 +245,18 @@ func TestClusterExtensionChannelVersionExists(t *testing.T) {
 	require.Equal(t, metav1.ConditionTrue, cond.Status)
 	require.Equal(t, ocv1alpha1.ReasonSuccess, cond.Reason)
 	require.Equal(t, "resolved to \"quay.io/operatorhubio/prometheus@fake1.0.0\"", cond.Message)
-	cond = apimeta.FindStatusCondition(clusterExtension.Status.Conditions, ocv1alpha1.TypeInstalled)
-	require.NotNil(t, cond)
-	require.Equal(t, metav1.ConditionUnknown, cond.Status)
-	require.Equal(t, ocv1alpha1.ReasonInstallationStatusUnknown, cond.Reason)
-	require.Equal(t, "bundledeployment status is unknown", cond.Message)
 
-	t.Log("By fetching the bundled deployment")
-	bd := &rukpakv1alpha2.BundleDeployment{}
-	require.NoError(t, cl.Get(ctx, types.NamespacedName{Name: extKey.Name}, bd))
-	require.Equal(t, "core-rukpak-io-registry", bd.Spec.ProvisionerClassName)
-	require.Equal(t, installNamespace, bd.Spec.InstallNamespace)
-	require.Equal(t, rukpakv1alpha2.SourceTypeImage, bd.Spec.Source.Type)
-	require.NotNil(t, bd.Spec.Source.Image)
-	require.Equal(t, "quay.io/operatorhubio/prometheus@fake1.0.0", bd.Spec.Source.Image.Ref)
+	// we're skipping based on same loadError since Status.Condition setting path is skipped
+	// and invariants will be different
+	if err != nil && err.Error() != utilerrors.NewAggregate([]error{loadError}).Error() {
+		cond = apimeta.FindStatusCondition(clusterExtension.Status.Conditions, ocv1alpha1.TypeInstalled)
```

Comment:
@bentito I think the whole test that `skipinvariants` is not necessary hereon - I remember chatting to @joelanford about this. Once we are able to remove it, I don't think we need to even mock the storage. The installationStatusUnknown here was because there was no rukpak running under the hood, and the firs reconcile would always return an unknown status because unpack was still in progress. 

So the test here should be to verify only till the state is pending. The objective of this unit test ended to ensure that resolve was successful and a bundle deployment was created with the installation status unknown. With our changes this translates to - Resolve status set to the right value and unpack status to be pending - which is what the first reconcile is suppose to return.

Reply:
This should solve all the unit follow up unit tests. Especially because these tests never ran rukpak previously - which means in current scenario all we need to test is till the unpack state is set to pending successfully.

Reply:
https://github.com/operator-framework/operator-controller/commit/d649164d840340ae475ba434caf4af2fb33ac801 - is what I meant. Doing this in the rest of the tests should solve all the cases. We shouldn't be needing to test loader or storer itself as that is something which these tests didn't do either - as they never ran rukpak.